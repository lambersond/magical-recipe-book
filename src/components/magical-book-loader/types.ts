export type MagicalBookLoaderProps = {
  loadingText?: string
  className?: string
}
