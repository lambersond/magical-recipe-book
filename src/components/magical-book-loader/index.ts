export { MagicalBookLoader } from './magical-book-loader'
