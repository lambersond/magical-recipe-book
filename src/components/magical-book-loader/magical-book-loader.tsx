import { MagicalBookLoaderProps } from './types'

export function MagicalBookLoader({
  loadingText = 'Brewing magical recipes...',
  className = '',
}: Readonly<MagicalBookLoaderProps>) {
  return (
    <div className={`magical-book-container ${className}`}>
      <div className='recipe-book'>
        {/* Book base structure */}
        <div className='book-base'></div>

        {/* Book pages background */}
        <div className='page-left-bg'></div>
        <div className='page-right-bg'></div>

        {/* Animated right page content */}
        <div className='right-page-content'>
          <div className='right-content-1'>
            <div className='squiggle-title'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line short'></div>
            <div className='squiggle-line medium'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line indent'></div>
            <div className='squiggle-line indent short'></div>
            <div className='image-placeholder'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line medium'></div>
            <div className='magical-symbol'>✧ ◆ ✧</div>
          </div>

          <div className='right-content-2'>
            <div className='squiggle-title'></div>
            <div className='squiggle-line medium'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line indent'></div>
            <div className='squiggle-line indent medium'></div>
            <div className='squiggle-line short'></div>
            <div className='image-placeholder'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line indent short'></div>
            <div className='squiggle-line medium'></div>
            <div className='magical-symbol'>◊ ⬟ ◊</div>
          </div>

          <div className='right-content-3'>
            <div className='squiggle-title'></div>
            <div className='image-placeholder'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line medium'></div>
            <div className='squiggle-line indent'></div>
            <div className='squiggle-line indent short'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line short'></div>
            <div className='squiggle-line medium'></div>
            <div className='squiggle-line'></div>
            <div className='magical-symbol'>❋ ◈ ❋</div>
          </div>

          <div className='right-content-4'>
            <div className='squiggle-title'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line short'></div>
            <div className='squiggle-line medium'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line indent'></div>
            <div className='squiggle-line indent short'></div>
            <div className='image-placeholder'></div>
            <div className='squiggle-line'></div>
            <div className='squiggle-line medium'></div>
            <div className='magical-symbol'>✧ ◆ ✧</div>
          </div>
        </div>

        <div className='spine'></div>

        {/* Flipping pages with squiggle content */}
        <div className='page page-1'>
          <div className='squiggle-title'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line medium'></div>
          <div className='squiggle-line short'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line indent'></div>
          <div className='squiggle-line indent short'></div>
          <div className='squiggle-line indent'></div>
          <div className='image-placeholder'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line medium'></div>
          <div className='magical-symbol'>⚡ ✨ ⚡</div>
        </div>

        <div className='page page-2'>
          <div className='squiggle-title'></div>
          <div className='image-placeholder'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line short'></div>
          <div className='squiggle-line indent'></div>
          <div className='squiggle-line indent medium'></div>
          <div className='squiggle-line indent short'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line medium'></div>
          <div className='squiggle-line short'></div>
          <div className='magical-symbol'>⟐ ◉ ⟐</div>
        </div>

        <div className='page page-3'>
          <div className='squiggle-title'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line medium'></div>
          <div className='squiggle-line indent'></div>
          <div className='squiggle-line indent short'></div>
          <div className='squiggle-line indent'></div>
          <div className='squiggle-line indent medium'></div>
          <div className='image-placeholder'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line short'></div>
          <div className='magical-symbol'>◯ ◉ ●</div>
        </div>

        <div className='page page-4'>
          <div className='squiggle-title'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line medium'></div>
          <div className='squiggle-line short'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line indent'></div>
          <div className='squiggle-line indent short'></div>
          <div className='squiggle-line indent'></div>
          <div className='image-placeholder'></div>
          <div className='squiggle-line'></div>
          <div className='squiggle-line medium'></div>
          <div className='magical-symbol'>⚡ ✨ ⚡</div>
        </div>

        {/* Magical sparkles */}
        <div className='sparkles'>
          <div className='sparkle'></div>
          <div className='sparkle'></div>
          <div className='sparkle'></div>
          <div className='sparkle'></div>
          <div className='sparkle'></div>
          <div className='sparkle'></div>
        </div>
      </div>

      <div className='loading-text'>{loadingText}</div>
    </div>
  )
}
