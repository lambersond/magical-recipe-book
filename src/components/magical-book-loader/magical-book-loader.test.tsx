import { render } from '@test-utils'
import { MagicalBookLoader } from './magical-book-loader'

describe('components/magical-book-loader', () => {
  it('matches snapshot', () => {
    const { container } = render(<MagicalBookLoader loadingText='Loading...' />)
    expect(container).toMatchSnapshot()
  })
})
