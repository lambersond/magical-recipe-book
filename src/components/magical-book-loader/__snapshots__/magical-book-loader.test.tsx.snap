// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/magical-book-loader matches snapshot 1`] = `
<div>
  <div
    class="magical-book-container "
  >
    <div
      class="recipe-book"
    >
      <div
        class="book-base"
      />
      <div
        class="page-left-bg"
      />
      <div
        class="page-right-bg"
      />
      <div
        class="right-page-content"
      >
        <div
          class="right-content-1"
        >
          <div
            class="squiggle-title"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line short"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line indent"
          />
          <div
            class="squiggle-line indent short"
          />
          <div
            class="image-placeholder"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="magical-symbol"
          >
            ✧ ◆ ✧
          </div>
        </div>
        <div
          class="right-content-2"
        >
          <div
            class="squiggle-title"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line indent"
          />
          <div
            class="squiggle-line indent medium"
          />
          <div
            class="squiggle-line short"
          />
          <div
            class="image-placeholder"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line indent short"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="magical-symbol"
          >
            ◊ ⬟ ◊
          </div>
        </div>
        <div
          class="right-content-3"
        >
          <div
            class="squiggle-title"
          />
          <div
            class="image-placeholder"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="squiggle-line indent"
          />
          <div
            class="squiggle-line indent short"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line short"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="magical-symbol"
          >
            ❋ ◈ ❋
          </div>
        </div>
        <div
          class="right-content-4"
        >
          <div
            class="squiggle-title"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line short"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line indent"
          />
          <div
            class="squiggle-line indent short"
          />
          <div
            class="image-placeholder"
          />
          <div
            class="squiggle-line"
          />
          <div
            class="squiggle-line medium"
          />
          <div
            class="magical-symbol"
          >
            ✧ ◆ ✧
          </div>
        </div>
      </div>
      <div
        class="spine"
      />
      <div
        class="page page-1"
      >
        <div
          class="squiggle-title"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line medium"
        />
        <div
          class="squiggle-line short"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="squiggle-line indent short"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="image-placeholder"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line medium"
        />
        <div
          class="magical-symbol"
        >
          ⚡ ✨ ⚡
        </div>
      </div>
      <div
        class="page page-2"
      >
        <div
          class="squiggle-title"
        />
        <div
          class="image-placeholder"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line short"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="squiggle-line indent medium"
        />
        <div
          class="squiggle-line indent short"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line medium"
        />
        <div
          class="squiggle-line short"
        />
        <div
          class="magical-symbol"
        >
          ⟐ ◉ ⟐
        </div>
      </div>
      <div
        class="page page-3"
      >
        <div
          class="squiggle-title"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line medium"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="squiggle-line indent short"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="squiggle-line indent medium"
        />
        <div
          class="image-placeholder"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line short"
        />
        <div
          class="magical-symbol"
        >
          ◯ ◉ ●
        </div>
      </div>
      <div
        class="page page-4"
      >
        <div
          class="squiggle-title"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line medium"
        />
        <div
          class="squiggle-line short"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="squiggle-line indent short"
        />
        <div
          class="squiggle-line indent"
        />
        <div
          class="image-placeholder"
        />
        <div
          class="squiggle-line"
        />
        <div
          class="squiggle-line medium"
        />
        <div
          class="magical-symbol"
        >
          ⚡ ✨ ⚡
        </div>
      </div>
      <div
        class="sparkles"
      >
        <div
          class="sparkle"
        />
        <div
          class="sparkle"
        />
        <div
          class="sparkle"
        />
        <div
          class="sparkle"
        />
        <div
          class="sparkle"
        />
        <div
          class="sparkle"
        />
      </div>
    </div>
    <div
      class="loading-text"
    >
      Loading...
    </div>
  </div>
</div>
`;
